docs/api-reference/cpp-api.rst

```diff
@@ -17,12 +17,136 @@ To search an API, refer to the API :ref:`genindex`.
 
 Device functions
 ================
+
+To use the device API, include the file ``rocrand_kernel.h`` in files that define kernels that use rocRAND device functions. The typical usage of device functions consists of the following operations in the device kernel definition:
+
+1. Create a new generator state object of the desired generator type.
+
+2. Initialize the generator state parameters using ``rocrand_init``.
+
+3. Generate random numbers by calling generation function on the generator state.
+
+4. Use the results.
+
+Since the rocRAND device functions are invoked from inside the user kernel, the generated numbers may be used right away in the kernel without a need to copy them to the host memory.
+
+In the below example, random number generation is using XORWOW generator.
+
+..  code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <rocrand/rocrand_kernel.h>
+    
+    __global__
+    void test()
+    {
+        uint                 tid = blockDim.x * blockIdx.x + threadIdx.x;
+        rocrand_state_xorwow state;
+        rocrand_init(123, tid, 0, &state);
+    
+        for(int i = 0; i < 3; ++i)
+        {
+            const auto value = rocrand(&state);
+            printf("thread %d, index %u: %u\n", tid, i, value);
+        }
+    }
+    
+    int main()
+    {
+        test<<<dim3(1), dim3(32)>>>();
+        hipDeviceSynchronize();
+    }
+
 .. doxygengroup:: rocranddevice
 
 C host API
 ==========
+
+C host API allows encapsulation of the internal generator state. Random numbers may be produced either on the host CPU or device GPU, whether an appropriate generator object was created. The typical sequence of operations for GPU generation consists of the following steps:
```

Comment:
Should this be "whether or not" or "depending on whether"? Right now, it is ambiguous.

Reply:
Reworded it in https://github.com/ROCm/rocRAND/pull/574/commits/9f8275c66b77a353c7c3cb51b6509b7cc009625b to better clarify that where (host/device) the random number generation is executed is dependent on the created generator.